### ISSUE TITLE
[HOLD for payment 2023-10-16] Chat - Expand button disappear after clicking even when composer has error

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#27779]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to any report
2. Paste the message that has over 15,000 characters (To make the error shows).
3. Now the composer will have red border and expand button is visible, click on the Send button.
4. Notice that now the Composer doesn’t have Expand button.

## Expected Result:
Expand button should be visible after clicking Send button

## Actual Result:
Expand button disappear after clicking even when composer has error

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.71.5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/ea69cd7e-99b2-47be-abcc-e918e153341f

https://github.com/Expensify/App/assets/93399543/4560eefb-e5fa-4c5f-8d6c-e6e5661a0f5b

**Expensify/Expensify Issue URL:**
**Issue reported by:** @hungvu193
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1694937956481939
https://expensify.slack.com/archives/C049HHMV9SM/p1695124790291359

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e5b7548a5c671a9e</li>
        <li>Upwork Job ID: 1704120906245259264</li>
        <li>Last Price Increase: 2023-09-19</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Contributor | 26978024</li>
<li>hungvu193 | Reporter | 26978028</li>
</ul></li>
    </ul>
</details>

**Comment by akinwale** [comment]
This has the same root cause as https://github.com/Expensify/App/issues/26960 where if the send button is set to disabled, the the tap action still gets processed.

**Comment by jliexpensify** [comment]
@robertKozik based on the comment above, this sounds like it could be a dupe? Shall we close this or maybe put this on hold?

**Comment by DylanDylann** [comment]
@jliexpensify I think ```App allows to write and send message in skeleton view``` is expected behavious

**Comment by robertKozik** [comment]
No, it's not a duplicate. The issue that @akinwale linked did contain a proposal with a solution that would also fix our issue. However, that issue as a whole was closed because it pertained to expected behavior. I'll review the proposals today.

**Comment by jliexpensify** [comment]
Any thoughts on the proposals @robertKozik ?

**Comment by robertKozik** [comment]
Hi @DylanDylann  @hungvu193! Both of you siggested simillar solution for this problem, which potencially suffers from the same problem. Per #26960 issue we want the send button to be accessible when chat is loading.
@hungvu193 your solution is passing the `isDisabledProp` which would disable the send button while loading
@DylanDylann is the `isSendDisabled` variable identical with `isDisabledProp`? If yes, it will case the same problem.

If you could update your proposal so that send button won't be disabled during chat loading, but be disabled only after clicking on it

**Comment by DylanDylann** [comment]
@robertKozik Also please note that my proposal will fix the case pressing the ENTER button, not only clicking on "Send" button

**Comment by DylanDylann** [comment]
@robertKozik 

> If you could update your proposal so that send button won't be disabled during chat loading, but be disabled only after clicking on it
You mean that the send button can be clicking on it (displayed as the green button) but with no action is triggered, right?

**Comment by robertKozik** [comment]
Thank you both for updating the proposals! I think I will suggest the @DylanDylann proposal, as it's not only fixes the issue, but as well refactor the code with unifying the submit functions.

Selected proposal: https://github.com/Expensify/App/issues/27779#issuecomment-1725481804

🎀 👀 🎀 C+ reviewed

**Comment by DylanDylann** [comment]
@robertKozik PR https://github.com/Expensify/App/pull/28681 is ready for review

**Comment by jliexpensify** [comment]
Should @DylanDylann be paid a reporting bonus here? It looks like there was a ~3 day delay for @robertKozik to review the PR cc @aldo-expensify for your thoughts

**Comment by robertKozik** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  [@robertKozik] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/25758
> * [ ]  [@robertKozik] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/25758/files#r1351875313
> * [ ]  [@robertKozik] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A**
> * [ ]  [@robertKozik] Determine if we should create a regression test for this bug. I don't think we need new test here
> * [ ]  [@robertKozik] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again. N/A

**Comment by DylanDylann** [comment]
@aldo-expensify Please help check https://github.com/Expensify/App/issues/27779#issuecomment-1752353186 when you have a chance

**Comment by jliexpensify** [comment]
Hi @DylanDylann - I actually queried this with another GH (same situation, delay in C+) and was told that bonuses only apply if the Expensify Engineer is delayed. Sorry!

**Comment by jliexpensify** [comment]
Payment Summary:

- Bug Reporter: @hungvu193 $50
- Contributor: @DylanDylann $500
- C+: @robertKozik (invoiced payment)

[Upwork job](https://www.upwork.com/jobs/~01e5b7548a5c671a9e)

**Comment by DylanDylann** [comment]
Yeah thanks @jliexpensify

**Comment by jliexpensify** [comment]
All paid and job closed!
